Exec/RegTests/SootRadTest/pelelmex_prob.H

```diff
@@ -0,0 +1,140 @@
+#ifndef PELELM_PROB_H
+#define PELELM_PROB_H
+
+#include <AMReX_Geometry.H>
+#include <AMReX_FArrayBox.H>
+#include <AMReX_ParmParse.H>
+
+#include <pelelmex_prob_parm.H>
+#include <PMF.H>
+#include <PMFData.H>
+
+#include <PeleLMeX_Index.H>
+#include <PelePhysics.H>
+#include "SootModel.H"
+
+AMREX_GPU_DEVICE
+AMREX_FORCE_INLINE
+void
+pelelmex_initdata(
+  int i,
+  int j,
+  int k,
+  int /*is_incompressible*/,
+  amrex::Array4<amrex::Real> const& state,
+  amrex::Array4<amrex::Real> const& aux,
+  amrex::GeometryData const& geomdata,
+  ProbParm const& prob_parm,
+  pele::physics::PMF::PmfData::DataContainer const* pmf_data)
+{
+
+  const amrex::Real* prob_lo = geomdata.ProbLo();
+  const amrex::Real* prob_hi = geomdata.ProbHi();
+  const amrex::Real* dx = geomdata.CellSize();
+
+  AMREX_D_TERM(const amrex::Real x = prob_lo[0] + (i + 0.5) * dx[0];
+               , const amrex::Real y = prob_lo[1] + (j + 0.5) * dx[1];
+               , const amrex::Real z = prob_lo[2] + (k + 0.5) * dx[2];);
+  amrex::GpuArray<amrex::Real, NUM_SPECIES> massfrac = {{0.0}};
+  amrex::GpuArray<amrex::Real, NUM_SPECIES + 4> pmf_vals = {{0.0}};
+  amrex::Real x1 = (x - prob_parm.standoff - 0.5 * dx[0]) * 100.;
+  amrex::Real x2 = (x - prob_parm.standoff + 0.5 * dx[0]) * 100.;
+  pele::physics::PMF::pmf(pmf_data, x1, x2, pmf_vals);
+  state(i, j, k, TEMP) = pmf_vals[1];
+  amrex::Real norm = 0.;
+  for (int n = 0; n < NUM_SPECIES; n++) {
+    massfrac[n] = amrex::max(0., amrex::min(1., pmf_vals[3 + n]));
+    norm += massfrac[n];
+  }
+  for (int n = 0; n < NUM_SPECIES; ++n) {
+    massfrac[n] = massfrac[n] / norm;
+  }
+  AMREX_D_TERM(state(i, j, k, VELX) = pmf_vals[0] * 1.E-2;
+               , state(i, j, k, VELY) = 0.;, state(i, j, k, VELZ) = 0.;);
+  amrex::Real rho_cgs, P_cgs;
+  P_cgs = prob_parm.P_mean * 10.;
```

Comment:
Avoid declaring and defining when you can do both: `const amrex::Real P_cgs = prob_parm.P_mean * 10.;`